### PR TITLE
Linter fixes

### LINKED ISSUE #59: Fix linter issues
**Description** 

We started without actually having linter and tests in our CI. When looking into the linter scripts, we could see that there are quite a few things to fix or warnings to check.

**Links / References**

- `pnpm run -r format` for formatting
- `pnpm run -r lint` for linting

**Tasks / Definition of Done**

- [x] Update all packages to eslint 9
- [x] Ensure linting works in all packages
- [ ] Fix linter warnings

**Further information**

This should be an automated check in the future for every pull request.

### PR DESCRIPTION
**Pull request by TamaraBogantseva**
**Checklist**

- [x] Resolves #59 
- [x] Labelled and assigned a reviewer

**Reviewer**

- [ ] I've checked out the code and tested it myself.

**Changes**

- _Update `eslint.config.mjs` and fix all issues._

### CODE REVIEW

File: flottform/forms/src/internal.ts [hunk 1]
```diff
@@ -51,13 +51,15 @@ export type FlottformState =
...
 
+/* eslint-disable @typescript-eslint/no-explicit-any */
 export type Logger = {
 	debug: (...args: any[]) => void;
 	info: (...args: any[]) => void;
 	log: (...args: any[]) => void;
 	warn: (...args: any[]) => void;
 	error: (...args: any[]) => void;
 };
+/* eslint-enable @typescript-eslint/no-explicit-any */
```
Comment (Narigo) [inline flottform/forms/src/internal.ts:62]: I think it's ok like this, but what do you think about `debug: typeof console.debug` etc? Would that work as well?

File: flottform/forms/src/internal.ts [hunk 2]
```diff
@@ -104,7 +106,7 @@ export function setIncludes<T>(set: Set<T>, x: T): boolean {
 	return false;
 }
 
-type Listener<T extends Array<any>> = (...args: T) => void;
+type Listener<T extends Array<any>> = (...args: T) => void; // eslint-disable-line @typescript-eslint/no-explicit-any
```
Comment (Narigo) [inline flottform/forms/src/internal.ts:109]: Not sure how to improve this, to be honest. Wondering how we could improve this type

File: flottform/server-standalone/src/routes/flottform/[endpointId]/client/+server.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ export const PUT: RequestHandler = async ({ params, request }) => {
 	let data: unknown;
 	try {
 		data = await request.json();
-	} catch (e) {
+	} catch {
```
Comment (Narigo) [inline flottform/server-standalone/src/routes/flottform/[endpointId]/client/+server.ts:21]: Maybe we should log the `e` for debug level or something?

**Review by Narigo — APPROVED** [review]
Thanks for taking this! Minor questions, can be merged and checked later, if you think it makes sense

File: flottform/forms/src/internal.ts (hunk 1 above)
Comment (TamaraBogantseva) [reply flottform/forms/src/internal.ts]: Yes, this also works. I changed it in new commit

File: flottform/forms/src/internal.ts (hunk 2 above)
Comment (TamaraBogantseva) [reply flottform/forms/src/internal.ts:107]: I'd create a new issue for this so we can have a closer look later 👍🏼

**Pull request merged**